backend/dataall/modules/dataset_sharing/services/share_notification_service.py

```diff
@@ -1,152 +1,140 @@
 import logging
+import enum
 from dataall.base.config import config
-from dataall.core.notifications.db.notification_repositories import Notification
-from dataall.core.notifications.db.notification_models import NotificationType
 from dataall.core.tasks.db.task_models import Task
 from dataall.core.tasks.service_handlers import Worker
 from dataall.modules.dataset_sharing.db.share_object_models import ShareObject
 from dataall.modules.datasets_base.db.dataset_models import Dataset
 from dataall.base.context import get_context
 from dataall.modules.dataset_sharing.db.enums import ShareObjectStatus
+from dataall.modules.notifications.db.notification_repositories import NotificationRepository
 
 log = logging.getLogger(__name__)
 
 
+class DataSharingNotificationType(enum.Enum):
+    SHARE_OBJECT_SUBMITTED = 'SHARE_OBJECT_SUBMITTED'
+    SHARE_ITEM_REQUEST = 'SHARE_ITEM_REQUEST'
+    SHARE_OBJECT_APPROVED = 'SHARE_OBJECT_APPROVED'
+    SHARE_OBJECT_REJECTED = 'SHARE_OBJECT_REJECTED'
+    SHARE_OBJECT_PENDING_APPROVAL = 'SHARE_OBJECT_PENDING_APPROVAL'
+    DATASET_VERSION = 'DATASET_VERSION'
+
+
 class ShareNotificationService:
-    @staticmethod
-    def notify_share_object_submission(
-            session, username: str, email_id: str, dataset: Dataset, share: ShareObject
-    ):
-        msg = f'User {username} submitted share request for dataset {dataset.label}'
-        subject = f'Data.all | Share Request Submitted for {dataset.label}'
-
-        notifications = [Notification.create(
-            session=session,
-            username=dataset.owner,
-            notification_type=NotificationType.SHARE_OBJECT_SUBMITTED,
-            target_uri=f'{share.shareUri}|{dataset.datasetUri}',
-            message=msg,
-        )]
-        session.add_all(notifications)
-
-        ShareNotificationService.create_notification_task(session, email_id, dataset, share, subject, msg)
+    """
+    Notifications sent to:
+        - dataset.SamlAdminGroupName
+        - dataset.stewards
+        - share.groupUri
```

Comment:
Looking at the way we set `self.notification_target_users = self._get_share_object_targeted_users()` it looks like we send notifications to 
        - dataset.SamlAdminGroupName
        - dataset.stewards
        - share.owner (NOT share.groupUri)

Reply:
Should we handle this similar to how we do for email (if group_notifications=true have it to share.groupUri and if not just to share.owner)?

Reply:
I had the same dilemma! My first thought was, let's keep both the same for consistency. It is what makes sense right? But then, looking at it from a different angle: the reason why we have the group notifications is because there will be extra emails to everyone, it is a matter of people being bothered with many emails. A small notification in the UI does not harm anyone and even if you are not "poked with an email" to attend the share request it gives users some visibility on it. I can change it because I am very undecided between the 2 options. @TejasRGitHub what is your opinion?

Reply:
Hi @noah-paige , @dlpzx , 

I agree we can have notifications based on the `group_notifications` settings for the UI as well.

Reply:
If we do this, i.e. have consistent logic for both UI and Email notifications then a single function _get_targeted_users... can be used for both the notification types

Reply:
I am going to leave them independent from each other. The final reason for this is that not all customers will use the `share_notifications` block and the `group_notifications` is a parameter specific to the email notifications